src/Utils.cc

```diff
@@ -28,6 +32,57 @@ namespace rendering
 {
 inline namespace IGNITION_RENDERING_VERSION_NAMESPACE {
 //
+/////////////////////////////////////////////////
+math::Vector3d ScreenToScene(
+    const math::Vector2i &_screenPos,
+    const CameraPtr &_camera,
+    const RayQueryPtr &_rayQuery)
+{
+  // Normalize point on the image
+  double width = _camera->ImageWidth();
+  double height = _camera->ImageHeight();
+
+  double nx = 2.0 * _screenPos.X() / width - 1.0;
+  double ny = 1.0 - 2.0 * _screenPos.Y() / height;
+
+  // Make a ray query
+  _rayQuery->SetFromCamera(
+      _camera, math::Vector2d(nx, ny));
+
+  auto result = _rayQuery->ClosestPoint();
+  if (result)
+    return result.point;
+
+  // Set point to be 10m away if no intersection found
+  return _rayQuery->Origin() +
+      _rayQuery->Direction() * 10;
+}
+
+/////////////////////////////////////////////////
+math::Vector3d ScreenToPlane(
+    const math::Vector2i &_screenPos,
+    const CameraPtr &_camera,
+    const RayQueryPtr &_rayQuery)
+{
+  // Normalize point on the image
+  double width = _camera->ImageWidth();
+  double height = _camera->ImageHeight();
+
+  double nx = 2.0 * _screenPos.X() / width - 1.0;
+  double ny = 1.0 - 2.0 * _screenPos.Y() / height;
+
+  // Make a ray query
+  _rayQuery->SetFromCamera(
+      _camera, math::Vector2d(nx, ny));
+
+  ignition::math::Planed plane(ignition::math::Vector3d(0, 0, 1), 0);
```

Comment:
can you make the plane elevation an optional argument to this `ScreenToPlane` function, with a default value of `0.0`?